app/uk/gov/hmrc/platforminitiatives/services/PlatformInitiativesService.scala

```diff
@@ -0,0 +1,115 @@
+
+  def allPlatformInitiatives(implicit ec: ExecutionContext): Future[Seq[PlatformInitiative]] = {
+    val initiatives = Seq(
+      createDefaultBranchInitiative(
+        initiativeName        = "Update Default Branch Terminology",
+        initiativeDescription = "To update default branch names - [Default branch tracker](https://catalogue.tax.service.gov.uk/defaultbranch).",
+        completedLegend       = "Updated",
+        inProgressLegend      = "Master"
+      ),
+      createUpgradeInitiative(
+        initiativeName        = "Play 2.6 upgrade",
+        initiativeDescription = "Play 2.6 upgrade - Deprecate [Play 2.5 and below](https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?pageId=275944511).",
+        dependencyName        = "play",
+        version               = Option(Version(2,6,0))
+      ),
+      createUpgradeInitiative(
+        initiativeName        = "Play 2.8 upgrade",
+        initiativeDescription = "Play 2.8 upgrade - Deprecate [Play 2.7 and below](https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?pageId=275944511).",
+        dependencyName        = "play",
+        version               = Option(Version(2,8,0))
+      ),
+      createUpgradeInitiative(
+        initiativeName        = "Auth-client upgrade",
+        initiativeDescription = "[CL250 Security upgrade required](https://confluence.tools.tax.service.gov.uk/x/RgpxDw)",
+        dependencyName        = "auth-client",
+        version               = Option(Version(5,6,0))
+      )
+    )
+    Future.sequence(initiatives)
+  }
+
+  def createDefaultBranchInitiative(
+     initiativeName              : String,
+     initiativeDescription       : String,
+     completedLegend             : String = "Completed",
+     inProgressLegend            : String = "Not Completed",
+   )(implicit ec: ExecutionContext): Future[PlatformInitiative] = {
+    teamsAndRepositoriesConnector.allDefaultBranches.map { repos =>
+      PlatformInitiative(
+        initiativeName            = initiativeName,
+        initiativeDescription     = initiativeDescription,
+        currentProgress           = repos
+          .filter   (!_.isArchived)
+          .map      (_.defaultBranch)
+          .count    (_ != "master"),
+        targetProgress            = repos
+          .filter   (!_.isArchived)
+          .map      (_.defaultBranch)
+          .length,
+        completedLegend           = completedLegend,
+        inProgressLegend          = inProgressLegend
+      )
+    }
+  }
+
+  def createUpgradeInitiative(
+    initiativeName              : String,
+    initiativeDescription       : String,
+    dependencyName              : String,
+    version                     : Option[Version],
+    completedLegend             : String = "Completed",
+    inProgressLegend            : String = "Not Completed",
+  )(implicit ec: ExecutionContext): Future[PlatformInitiative] = {
+      serviceDependenciesConnector.getAllDependencies.map { repos =>
```

Comment:
since you're getting all dependencies, this could be done once and the result passed to each call to `createUpgradeInitiative`? Alterntively (and probably simpler) add a cache into `getAllDependencies` (e.g. https://github.com/hmrc/service-dependencies/blob/3a439926519e2e388b7878cc2cf163a96b7cd374/app/uk/gov/hmrc/servicedependencies/connector/ServiceConfigsConnector.scala#L46)

Also rename param from `repos` to `dependencies`